app/services/course/assessment/answer/text_response_auto_grading_service.rb

```diff
@@ -1,8 +1,13 @@
 # frozen_string_literal: true
+require 'rwordnet'
 class Course::Assessment::Answer::TextResponseAutoGradingService < \
   Course::Assessment::Answer::AutoGradingService
   def evaluate(answer)
-    answer.correct, grade, messages = evaluate_answer(answer.actable)
+    if answer.question.actable.comprehension_question?
+      answer.correct, grade, messages = evaluate_answer_comprehension(answer.actable)
```

Comment:
Can you split all the new logic into a separate service, and call it here? So it should be something like: 
```ruby
answer.correct, grade, messages = Course::Assessment::Answer::TextResponseComprehensionAutoGradingService.evaluate(answer.actable)
```

Reply:
Okay, will do.